pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -236,17 +263,75 @@ func (r *reconcilerBase) createOrPatchDeployment(ctx context.Context, declared *
 		r.log.Error(err, "Managed object update failed, deleting and re-creating",
 			logFieldObject, dRef.String(),
 			logFieldKind, kind)
-		if err := r.client.Delete(ctx, declared); err != nil {
+		if err := deploymentResourceInterface.Delete(ctx, dRef.Name, metav1.DeleteOptions{}); err != nil {
```

Comment:
Do we want to be using the Foreground PropagationPolicy here? That would ensure the ReplicaSets & Pods are deleted before we continue.